spring-integration-ip/src/main/java/org/springframework/integration/ip/config/IpAdapterParserUtils.java

```diff
@@ -153,12 +155,16 @@ public static void addConstuctorValueIfAttributeDefined(BeanDefinitionBuilder bu
 	public static void addHostAndPortToConstructor(Element element,
 			BeanDefinitionBuilder builder, ParserContext parserContext) {
 		String host = element.getAttribute(IpAdapterParserUtils.HOST);
-		if (!StringUtils.hasText(host)) {
+		String socketExpression = element.getAttribute(IpAdapterParserUtils.SOCKET_EXPRESSION);
+		if (!StringUtils.hasText(host) && StringUtils.isEmpty(socketExpression)) {
 			parserContext.getReaderContext().error(IpAdapterParserUtils.HOST
 					+ " is required for IP outbound channel adapters", element);
 		}
 		builder.addConstructorArgValue(host);
 		String port = IpAdapterParserUtils.getPort(element, parserContext);
+		if (StringUtils.isEmpty(port)) {
+			port = "0";
```

Comment:
Fixed.
Actually `socket-expression` is mutually exclusive with any of: host, port, multicast, acknowledge.